crates/matrix-sdk-base/src/sliding_sync/http.rs

```diff
@@ -12,38 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! HTTP types for MSC4186 or MSC3585.
+//! HTTP types for MSC4186.
 //!
-//! This module provides unified namings for types from MSC3575 and
-//! MSC4186.
+//! It's a re-export of the type from Ruma.
 
-/// HTTP types from MSC3575, renamed to match the MSC4186 namings.
-pub mod msc3575 {
-    use ruma::api::client::sync::sync_events::v4;
-    pub use v4::{Request, Response};
-
-    /// HTTP types related to a `Request`.
-    pub mod request {
-        pub use super::v4::{
-            AccountDataConfig as AccountData, ExtensionsConfig as Extensions,
-            ReceiptsConfig as Receipts, RoomDetailsConfig as RoomDetails, RoomSubscription,
-            SyncRequestList as List, SyncRequestListFilters as ListFilters,
-            ToDeviceConfig as ToDevice, TypingConfig as Typing,
-        };
-    }
-
-    /// HTTP types related to a `Response`.
-    pub mod response {
-        pub use super::v4::{
-            AccountData, Extensions, Receipts, SlidingSyncRoom as Room,
-            SlidingSyncRoomHero as RoomHero, SyncList as List, ToDevice, Typing,
-        };
-    }
-}
-
-/// HTTP types from MSC4186.
-pub mod msc4186 {
-    pub use ruma::api::client::sync::sync_events::v5::*;
-}
-
-pub use msc4186::*;
+pub use ruma::api::client::sync::sync_events::v5::*;
```

Comment:
Can we get rid of this file now, and just use `use` with the ruma full path?

Reply:
I've removed this file, but `ruma::api::client::sync::sync_events::v5` is re-exported as `http` inside `matrix_sdk_base::sliding_sync`. I think it is still comfortable to use `matrix_sdk_base::sliding_sync::http` to get all the HTTP types for sliding sync without having to look for inside Ruma. It's just a module alias. What do you think?

Reply:
Hmm, I do get it's to avoid more changes that you're making this alias public, but if it's now part of our public API, this has semantic versioning implications: any changes to the underlying data structures becomes a breaking change in our public API, that we should mention, be aware of, etc.

Plus, it's another layer of indirection when browsing the code: if I look at a type `http::Something`, I may now think it's defined somewhere along here, find where the `http` declaration is, and find out it's a renaming from another module.

I'd rather not do that, if we can avoid it, and use the full paths in all places we use those types, making it super explicit and direct and non-ambiguous and not in our public API.

Reply:
As you prefer Sir.

Reply:
Done.